### PR TITLE
Openvswitch bridge issue fix #51480

### LINKED ISSUE #51480: openvswitch_bridge external_ids TypeError: argument of type 'NoneType' is not iterable
##### SUMMARY
When external_ids is mentioned openvswitch_bridge module throws error. Without giving args it works fine.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
```
---
- name: Node Join Checkpoint Start
  hosts: all
  gather_facts: false
  tasks:
  - name: NSX | Create OVS bridge
    openvswitch_bridge:
      bridge: br-int
      state: present
      fail_mode: standalone
    args:
     external_ids:
        bridge-id: "br-int"
```
##### ANSIBLE VERSION

```
[root@sc-rdops-vm14-dhcp-178-105 myplay]# ansible --version
ansible 2.6.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May  3 2017, 07:55:04) [GCC 4.8.5 20150623 (Red Hat 4.8.5-14)]
```

##### CONFIGURATION

```
# ansible-config dump --only-changed
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles']
```

##### STEPS TO REPRODUCE

Run below playbook

```
---
- name: Node Join Checkpoint Start
  hosts: all
  gather_facts: false
  tasks:
  - name: NSX | Create OVS bridge
    openvswitch_bridge:
      bridge: br-int
      state: present
      fail_mode: standalone
    args:
     external_ids:
        bridge-id: "br-int"
```

##### ACTUAL RESULTS

```
# ansible-playbook -i hosts myplay/test.yml

PLAY [Node Join Checkpoint Start] ********************************************************************************************************************************************************

TASK [NSX | Create OVS bridge] ***********************************************************************************************************************************************************
fatal: [sc-rdops-vm14-dhcp-174-212.eng.vmware.com]: FAILED! => {"changed": false, "module_stderr": "Shared connection to sc-rdops-vm14-dhcp-174-212.eng.vmware.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_UnIT54/ansible_module_openvswitch_bridge.py\", line 267, in <module>\r\n    main()\r\n  File \"/tmp/ansible_UnIT54/ansible_module_openvswitch_bridge.py\", line 254, in main\r\n    commands = map_obj_to_commands(want, have, module)\r\n  File \"/tmp/ansible_UnIT54/ansible_module_openvswitch_bridge.py\", line 139, in map_obj_to_commands\r\n    if (k not in have['external_ids']\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "rc": 1}
fatal: [sc-rdops-vm14-dhcp-189-249.eng.vmware.com]: FAILED! => {"changed": false, "module_stderr": "Shared connection to sc-rdops-vm14-dhcp-189-249.eng.vmware.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_UC6B61/ansible_module_openvswitch_bridge.py\", line 267, in <module>\r\n    main()\r\n  File \"/tmp/ansible_UC6B61/ansible_module_openvswitch_bridge.py\", line 254, in main\r\n    commands = map_obj_to_commands(want, have, module)\r\n  File \"/tmp/ansible_UC6B61/ansible_module_openvswitch_bridge.py\", line 139, in map_obj_to_commands\r\n    if (k not in have['external_ids']\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "rc": 1}
fatal: [sc-rdops-vm14-dhcp-178-105.eng.vmware.com]: FAILED! => {"changed": false, "module_stderr": "Shared connection to sc-rdops-vm14-dhcp-178-105.eng.vmware.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_mSapdO/ansible_module_openvswitch_bridge.py\", line 267, in <module>\r\n    main()\r\n  File \"/tmp/ansible_mSapdO/ansible_module_openvswitch_bridge.py\", line 254, in main\r\n    commands = map_obj_to_commands(want, have, module)\r\n  File \"/tmp/ansible_mSapdO/ansible_module_openvswitch_bridge.py\", line 139, in map_obj_to_commands\r\n    if (k not in have['external_ids']\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "rc": 1}
	to retry, use: --limit @/root/myplay/test.retry

PLAY RECAP *******************************************************************************************************************************************************************************
sc-rdops-vm14-dhcp-174-212.eng.vmware.com : ok=0    changed=0    unreachable=0    failed=1
sc-rdops-vm14-dhcp-178-105.eng.vmware.com : ok=0    changed=0    unreachable=0    failed=1
sc-rdops-vm14-dhcp-189-249.eng.vmware.com : ok=0    changed=0    unreachable=0    failed=1
```

### PR DESCRIPTION
**Pull request by vishwanathjadhav**
**SUMMARY:**

Exception while creating a bridge port, due to None type item.

Issue fix #51480

**ISSUE TYPE:**
- Bugfix Pull Request

**COMPONENT NAME:**

Open vSwitch

**ADDITIONAL INFORMATION:**

### CODE REVIEW

**Comment by vishwanathjadhav** [comment]
**The issue has been reproduced:**

~~~
TASK [Create bridge br0] *******************************************************
fatal: [sc-rdops-vm14-dhcp-174-212.eng.vmware.com]: FAILED! => {"changed": false, "module_stderr": "Shared connection to sc-rdops-vm14-dhcp-174-212.eng.vmware.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_UnIT54/ansible_module_openvswitch_bridge.py\", line 267, in <module>\r\n    main()\r\n  File \"/tmp/ansible_UnIT54/ansible_module_openvswitch_bridge.py\", line 254, in main\r\n    commands = map_obj_to_commands(want, have, module)\r\n  File \"/tmp/ansible_UnIT54/ansible_module_openvswitch_bridge.py\", line 139, in map_obj_to_commands\r\n    if (k not in have['external_ids']\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "rc": 1}
fatal: [sc-rdops-vm14-dhcp-189-249.eng.vmware.com]: FAILED! => {"changed": false, "module_stderr": "Shared connection to sc-rdops-vm14-dhcp-189-249.eng.vmware.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_UC6B61/ansible_module_openvswitch_bridge.py\", line 267, in <module>\r\n    main()\r\n  File \"/tmp/ansible_UC6B61/ansible_module_openvswitch_bridge.py\", line 254, in main\r\n    commands = map_obj_to_commands(want, have, module)\r\n  File \"/tmp/ansible_UC6B61/ansible_module_openvswitch_bridge.py\", line 139, in map_obj_to_commands\r\n    if (k not in have['external_ids']\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "rc": 1}
fatal: [sc-rdops-vm14-dhcp-178-105.eng.vmware.com]: FAILED! => {"changed": false, "module_stderr": "Shared connection to sc-rdops-vm14-dhcp-178-105.eng.vmware.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_mSapdO/ansible_module_openvswitch_bridge.py\", line 267, in <module>\r\n    main()\r\n  File \"/tmp/ansible_mSapdO/ansible_module_openvswitch_bridge.py\", line 254, in main\r\n    commands = map_obj_to_commands(want, have, module)\r\n  File \"/tmp/ansible_mSapdO/ansible_module_openvswitch_bridge.py\", line 139, in map_obj_to_commands\r\n    if (k not in have['external_ids']\r\nTypeError: argument of type 'NoneType' is not iterable\r\n", "msg": "MODULE FAILURE", "rc": 1}
	to retry, use: --limit @/root/myplay/test.retry
~~~


**Result After applying fix:**

~~~
TASK [Create bridge br0] *******************************************************
changed: [rhel7] => {"changed": true, "commands": ["/bin/ovs-vsctl -t 5 add-br br0", "/bin/ovs-vsctl -t 5 set-fail-mode br0 standalone", "/bin/ovs-vsctl -t 5 br-set-external-id br0 bridge-id br0"]}

TASK [Add interface to br0] ****************************************************
changed: [rhel7] => {"changed": true, "cmd": ["ovs-vsctl", "add-port", "br0", "enp0s8"], "delta": "0:00:00.007307", "end": "2019-06-19 06:35:37.373628", "rc": 0, "start": "2019-06-19 06:35:37.366321", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Create VXLAN] ************************************************************
changed: [rhel7] => {"changed": true, "cmd": ["ovs-vsctl", "add-port", "br0", "vxlan1", "--", "set", "interface", "vxlan1", "type=vxlan", "options:local_ip=192.168.11.2"], "delta": "0:00:00.006558", "end": "2019-06-19 06:35:37.724789", "rc": 0, "start": "2019-06-19 06:35:37.718231", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

PLAY RECAP *********************************************************************
rhel7                      : ok=12   changed=10   unreachable=0    failed=0    skipped=0    rescued=0    ignored=1   
~~~

**Sample Playbook to reprodue the issue and test the fix:**

~~~
- name: Reproduce and fix OVS issue
  hosts: "{{HOSTS}}"
  become: true
  vars:
  tasks:
  - when: ansible_distribution_major_version == '7'
    block:
    - name: Install OpenvSwitch
      yum:
        name: openvswitch
        state: latest

    - name: Restart OpenvSwitch
      service:
        name: "{{item}}"
        state: restarted
      ignore_errors: true
      with_items:
        - network
        - openvswitch
    
    - name: Create bridge br0
      openvswitch_bridge:
        bridge: br0
        state: present
        fail_mode: standalone
      args:
        external_ids:
            bridge-id: "br0"

    - name: Add interface to br0
      command: ovs-vsctl add-port br0 enp0s8
      ignore_errors: true

    - name: Create VXLAN
      command: ovs-vsctl add-port br0 vxlan1 -- set interface vxlan1 type=vxlan options:local_ip=192.168.11.2
      ignore_errors: true
~~~

File: lib/ansible/modules/network/ovs/openvswitch_bridge.py [hunk 1]
```diff
@@ -134,7 +134,8 @@ def map_obj_to_commands(want, have, module):
                 templatized_command = ("%(ovs-vsctl)s -t %(timeout)s"
                                        " br-set-external-id %(bridge)s")
                 command = templatized_command % module.params
-                if want['external_ids']:
+                if 'external_ids' in want and want['external_ids'] and\
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ovs/openvswitch_bridge.py]: The condition can be simplified as 
`if want.get('external_ids') and have.get('external_ids')`

**Review by pabelanger — CHANGES_REQUESTED** [review]
Thanks for the PR, lets also update test/integration/targets/openvswitch_bridge/ to include your example, to ensure we get test coverage.

File: lib/ansible/modules/network/ovs/openvswitch_bridge.py (hunk 1 above)
Comment (vishwanathjadhav) [reply lib/ansible/modules/network/ovs/openvswitch_bridge.py]: Done

**Comment by vishwanathjadhav** [comment]
@pabelanger - I have added the test case please have a look.

**Comment by danielmellado** [comment]
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/openvswitch.openvswitch
Please re-submit this PR in the above repository.
If you have further questions please stop by IRC or the mailing list:

IRC: #ansible-network on irc.freenode.net
mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**